src/components/views/rooms/RoomHeader.js

```diff
@@ -146,89 +146,84 @@ module.exports = React.createClass({
-            );
+            canSetRoomName = this.props.room.currentState.maySendStateEvent('m.room.name', userId);
+            canSetRoomAvatar = this.props.room.currentState.maySendStateEvent('m.room.avatar', userId);
+            canSetRoomTopic = this.props.room.currentState.maySendStateEvent('m.room.topic', userId);
 
-            save_button = (
+            saveButton = (
                 <AccessibleButton className="mx_RoomHeader_textButton" onClick={this.props.onSaveClick}>
                     {_t("Save")}
                 </AccessibleButton>
             );
         }
 
         if (this.props.onCancelClick) {
-            cancel_button = <CancelButton onClick={this.props.onCancelClick}/>;
+            cancelButton = <CancelButton onClick={this.props.onCancelClick}/>;
         }
 
         if (this.props.saving) {
-            var Spinner = sdk.getComponent("elements.Spinner");
+            const Spinner = sdk.getComponent("elements.Spinner");
             spinner = <div className="mx_RoomHeader_spinner"><Spinner/></div>;
         }
 
-        if (can_set_room_name) {
-            var RoomNameEditor = sdk.getComponent("rooms.RoomNameEditor");
+        if (canSetRoomName) {
+            const RoomNameEditor = sdk.getComponent("rooms.RoomNameEditor");
             name = <RoomNameEditor ref="nameEditor" room={this.props.room} />;
-        }
-        else {
-            var searchStatus;
+        } else {
             // don't display the search count until the search completes and
             // gives us a valid (possibly zero) searchCount.
-            if (this.props.searchInfo && this.props.searchInfo.searchCount !== undefined && this.props.searchInfo.searchCount !== null) {
-                searchStatus = <div className="mx_RoomHeader_searchStatus">&nbsp;{ _t("(~%(count)s results)", { count: this.props.searchInfo.searchCount }) }</div>;
+            if (this.props.searchInfo &&
+                this.props.searchInfo.searchCount !== undefined &&
+                this.props.searchInfo.searchCount !== null) {
+                searchStatus = <div className="mx_RoomHeader_searchStatus">&nbsp;
+                    { _t("(~%(count)s results)", { count: this.props.searchInfo.searchCount }) }
+                </div>;
             }
 
             // XXX: this is a bit inefficient - we could just compare room.name for 'Empty room'...
-            var settingsHint = false;
-            var members = this.props.room ? this.props.room.getJoinedMembers() : undefined;
+            let settingsHint = false;
+            const members = this.props.room ? this.props.room.getJoinedMembers() : undefined;
             if (members) {
                 if (members.length === 1 && members[0].userId === MatrixClientPeg.get().credentials.userId) {
-                    var name = this.props.room.currentState.getStateEvents('m.room.name', '');
+                    name = this.props.room.currentState.getStateEvents('m.room.name', '');
```

Comment:
`name` ends up being rendered in the jsx, so this doesn't look right. It might be best to rename all usages of `name`